plugins/client/graphql-kotlin-client-generator/src/main/kotlin/com/expediagroup/graphql/plugin/client/generator/types/generateTypeName.kt

```diff
@@ -258,3 +300,47 @@ private fun calculateSelectedFields(
     }
     return result
 }
+
+/**
+ * Determines if a GraphQL object type should be created as a shared response type.
+ * This checks if the feature is enabled and if the type is used in multiple operations.
+ */
+private fun shouldCreateSharedResponseType(context: GraphQLClientGeneratorContext, typeName: String): Boolean {
+    // Only create shared types if the feature is enabled
+    if (!context.config.useSharedResponseTypes) {
+        return false
+    }
+
+    // Use usage-based detection: create shared types for types used in multiple operations
+    val usageCount = context.typeUsageCount[typeName] ?: 0
```

Comment:
Correct me if I am wrong, but the `typeUsageTracker` logic purpose is just to know if a type is being used more than once ? the whole `typeUsageTracker` logic, could be avoided if we just treat all types as candidates for reusability, `analyzeTypeUsage` is a very expensive operation in terms of CPU, client generation does not happen on runtime, still It could slow down build times.